core/modules/comment/comment.admin.inc

```diff
@@ -6,217 +6,45 @@
-  $header = array(
-    'subject' => array('data' => t('Title'), 'field' => 'subject'),
-    'author' => array('data' => t('Author'), 'field' => 'name', 'class' => array(RESPONSIVE_PRIORITY_MEDIUM)),
-    'posted_in' => array('data' => t('Posted in'), 'field' => 'node_title', 'class' => array(RESPONSIVE_PRIORITY_LOW)),
-    'changed' => array('data' => t('Updated'), 'field' => 'c.changed', 'sort' => 'desc', 'class' => array(RESPONSIVE_PRIORITY_LOW)),
-    'operations' => t('Operations'),
-  );
-
-  $query = db_select('comment', 'c')->extend('PagerDefault')->extend('TableSort');
-  $query->join('node', 'n', 'n.nid = c.nid');
-  $query->addField('n', 'title', 'node_title');
-  $query->addTag('node_access');
-  $result = $query
-    ->fields('c', array('cid', 'subject', 'name', 'changed'))
-    ->condition('c.status', $status)
-    ->limit(50)
-    ->orderByHeader($header)
-    ->execute();
-
-  $cids = array();
-
-  // We collect a sorted list of node_titles during the query to attach to the
-  // comments later.
-  foreach ($result as $row) {
-    $cids[] = $row->cid;
-    $node_titles[] = $row->node_title;
+  else {
+    $cids = array();
   }
-  $comments = comment_load_multiple($cids);
 
-  // Build a table listing the appropriate comments.
-  $options = array();
-  $destination = backdrop_get_destination();
+  $form['#tree'] = TRUE;
 
-  foreach ($comments as $comment) {
-    // Remove the first node title from the node_titles array and attach to
-    // the comment.
-    $comment->node_title = array_shift($node_titles);
-    $options[$comment->cid] = array(
-      'subject' => array(
-        'data' => array(
-          '#type' => 'link',
-          '#title' => $comment->subject,
-          '#href' => 'comment/' . $comment->cid,
-          '#options' => array('attributes' => array('title' => truncate_utf8($comment->comment_body[LANGUAGE_NONE][0]['value'], 128)), 'fragment' => 'comment-' . $comment->cid),
-        ),
-      ),
-      'author' => theme('username', array('account' => $comment)),
-      'posted_in' => array(
-        'data' => array(
-          '#type' => 'link',
-          '#title' => $comment->node_title,
-          '#href' => 'node/' . $comment->nid,
-        ),
-      ),
-      'changed' => format_date($comment->changed, 'short'),
-    );
-    $links = array();
-    $links['edit'] = array(
-      'title' => t('Edit'),
-      'href' => 'comment/' . $comment->cid . '/edit',
-      'query' => $destination,
-    );
-    $options[$comment->cid]['operations']['data'] = array(
-      '#type' => 'operations',
-      '#links' => $links,
-    );
+  if (empty($cids)) {
+    $destination = isset($_GET['destination']) ? $_GET['destination'] : 'admin/content/comment';
+    $form['empty']['#markup'] = '<p>' . t('Return to the <a href="!url">comment administration page</a>.', array('!url' => url($destination))) . '</p>';
+    backdrop_set_message(t('No comments has been selected for deletion.'), 'error');
```

Comment:
either plural `comments` and `have`, or singular and `has`

Reply:
plural + have.